# HG changeset patch
# User Ben Abbott <bpabbott@mac.com>
# Date 1252539689 14400
#      Wed Sep 09 19:41:29 2009 -0400
# Node ID 40acd13920e36f2e0ac9318cf9f96017ecd8a515
# Parent  00958d0c4e3cf232305d184c5d519906633a1447
gnuplot_drawnow.m: Avoid flickering plot windows.

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,14 @@
+2009-09-09  Tatsuro Matsuoka <tmacchant@yahoo.co.jp>
+
+	* plot/gnuplot_drawnow.m: Avoid flickering windows by avoding
+	'set multiplot' / 'unset multiplot' for gnuplot's windows and
+	wxt terminals.
+
 2009-09-06  Jaroslav Hajek  <highegg@gmail.com>
 
 	* optimization/__dogleg__.m: Revert to revision 22c8272af34b.
 	* optimization/__doglegm__.m: New source.
 	* optimization/fminunc.m: Use it.
 
 2009-09-08  John W. Eaton  <jwe@octave.org>
 
diff --git a/scripts/plot/gnuplot_drawnow.m b/scripts/plot/gnuplot_drawnow.m
--- a/scripts/plot/gnuplot_drawnow.m
+++ b/scripts/plot/gnuplot_drawnow.m
@@ -283,30 +283,32 @@ function [enhanced, implicit_margin] = g
       ## Options must go last.
       term_str = sprintf ("%s %s", term_str, opts_str);
     endif
     if (! isempty (size_str) && new_stream)
       ## size_str comes after other options to permit specification of
       ## the canvas size for terminals cdr/corel.
       term_str = sprintf ("%s %s", term_str, size_str);
     endif
-    ## Work around the gnuplot feature of growing the x11 window when
-    ## the mouse and multiplot are set.
+    ## Work around the gnuplot feature of growing the x11 window and
+    ## flickering window (x11, windows, & wxt) when the mouse and
+    ## multiplot are set in gnuplot.
     fputs (plot_stream, "unset multiplot;\n");
-    if (! strcmp (term, "x11")
+    flickering_terms = {"x11", "windows", "wxt"};
+    if (! any (strcmp (term, flickering_terms))
         || numel (findall (h, "type", "axes")) > 1
         || numel (findall (h, "type", "image")) > 0)
       fprintf (plot_stream, "%s\n", term_str);
       if (nargin == 5)
         if (! isempty (file))
           fprintf (plot_stream, "set output '%s';\n", file);
         endif
       endif
       fputs (plot_stream, "set multiplot;\n");
-    elseif (strcmp (term, "x11"))
+    elseif (any (strcmp (term, flickering_terms)))
       fprintf (plot_stream, "%s\n", term_str);
       if (nargin == 5)
         if (! isempty (file))
           fprintf (plot_stream, "set output '%s';\n", file);
         endif
       endif
     endif
   else
